solidity/test/bridge/Bridge.Wallets.test.ts

```diff
@@ -0,0 +1,629 @@
+  // Fund the `walletRegistry` account so it's possible to mock sending requests
+  // from it.
+  await deployer.sendTransaction({
+    to: walletRegistry.address,
+    value: ethers.utils.parseEther("1"),
+  })
+
+  const BitcoinTx = await ethers.getContractFactory<BitcoinTx__factory>(
+    "BitcoinTx"
+  )
+  const bitcoinTx = await BitcoinTx.deploy()
+  await bitcoinTx.deployed()
+
+  const Wallets = await ethers.getContractFactory<Wallets__factory>("Wallets")
+  const wallets = await Wallets.deploy()
+  await wallets.deployed()
+
+  const Bridge = await ethers.getContractFactory<BridgeStub__factory>(
+    "BridgeStub",
+    {
+      libraries: {
+        BitcoinTx: bitcoinTx.address,
+        Wallets: wallets.address,
+      },
+    }
+  )
+  const bridge: Bridge & BridgeStub = await Bridge.deploy(
+    bank.address,
+    relay.address,
+    treasury.address,
+    walletRegistry.address,
+    1
+  )
+  await bridge.deployed()
+
+  await bank.updateBridge(bridge.address)
+  await bridge.connect(deployer).transferOwnership(governance.address)
+
+  return {
+    governance,
+    thirdParty,
+    treasury,
+    bank,
+    relay,
+    walletRegistry,
+    bridge,
+  }
+}
+
+describe("Bridge - Wallets", () => {
+  let governance: SignerWithAddress
+  let thirdParty: SignerWithAddress
+
+  let walletRegistry: FakeContract<IWalletRegistry>
+  let bridge: Bridge & BridgeStub
+
+  before(async () => {
+    // eslint-disable-next-line @typescript-eslint/no-extra-semi
+    ;({ governance, thirdParty, walletRegistry, bridge } =
+      await waffle.loadFixture(fixture))
+  })
+
+  describe("requestNewWallet", () => {
+    before(async () => {
+      await createSnapshot()
+    })
+
+    after(async () => {
+      walletRegistry.requestNewWallet.reset()
+
+      await restoreSnapshot()
+    })
+
+    context("when called by a third party", async () => {
+      context("when wallet creation is not in progress", () => {
+        before(async () => {
+          await createSnapshot()
+
+          walletRegistry.getWalletCreationState.returns(ecdsaDkgState.IDLE)
+        })
+
+        after(async () => {
+          walletRegistry.getWalletCreationState.reset()
+
+          await restoreSnapshot()
+        })
+
+        context("when active wallet is not set", () => {
+          let tx: ContractTransaction
+
+          before(async () => {
+            await createSnapshot()
+
+            tx = await bridge.connect(thirdParty).requestNewWallet(NO_MAIN_UTXO)
+          })
+
+          after(async () => {
+            walletRegistry.requestNewWallet.reset()
+
+            await restoreSnapshot()
+          })
+
+          it("should emit NewWalletRequested event", async () => {
+            await expect(tx).to.emit(bridge, "NewWalletRequested")
+          })
+
+          it("should call ECDSA Wallet Registry's requestNewWallet function", async () => {
+            await expect(walletRegistry.requestNewWallet).to.have.been
```

Comment:
This `await` is probably not needed.

Reply:
See https://github.com/keep-network/tbtc-v2/pull/172/commits/8f091ed3ca6c2ed76f61207291b5af75a05de3f2